executor/executor.go

```diff
@@ -328,7 +333,96 @@ func (executor *Executor) execute(
 			}
 			return response, nil
 		}
-		msg := "decision executed successfully"
+
+
+		// short pooling to trigger pod status with max 15 minutes
+		statusMap := make(map[kv1.PodPhase]string)
+		statusMap[kv1.PodRunning] = "pods restarted successfully"
+		statusMap[kv1.PodFailed] = "pods failed to restart"
+		statusMap[kv1.PodUnknown] = "pods status is unknown"
+
+		msg := "pods restarting exceeded timout (15 min)"
+		start := time.Now()
+
+		entitiName := ""
+		var targetPodCount int32 = 0
+		var runningPods int32 = 0
+		flag := false
+
+		if kind == "Deployment"{
+
+			replicasets, err := executor.kube.GetNamespaceReplicaSets(namespace)
+
+			if err != nil {
+				flag = true
+
+			}else{
+				// get the new replicaset
+				for _, replica := range replicasets.Items {
+					if strings.Contains(replica.Name, name) && replica.Status.ReadyReplicas > 0{
```

Comment:
Please make sure that you check the last rs status not just any rs with the same name.

You can either sort by `creationTimestamp` from the api server or just get the last rs that have the latest `creationTimestamp` and make sure that it's ready replicas is > 0

the `creationTimestamp` is within the `metadata`.

```
    "metadata": {
        "annotations": {
            "deployment.kubernetes.io/desired-replicas": "1",
            "deployment.kubernetes.io/max-replicas": "2",
            "deployment.kubernetes.io/revision": "98"
        },
        "creationTimestamp": "2020-05-12T18:43:51Z",
...
}

```

@mohamedveron

Reply:
i fetch the ready one and the api itself list the replicas sorted with the latest